dotnet/src/Agents/OpenAI/Internal/AssistantThreadActions.cs

```diff
@@ -111,31 +109,26 @@ public static async IAsyncEnumerable<ChatMessageContent> GetMessagesAsync(Assist
     {
         Dictionary<string, string?> agentNames = []; // Cache agent names by their identifier
 
-        await foreach (PageResult<ThreadMessage> page in client.GetMessagesAsync(threadId, new() { Order = ListOrder.NewestFirst }, cancellationToken).ConfigureAwait(false))
+        await foreach (var message in client.GetMessagesAsync(threadId, new() { Order = MessageCollectionOrder.Descending }, cancellationToken).ConfigureAwait(false))
```

Comment:
Looks like the API has gone back and forth on this.  Ultimately, I question if not supporting paging will be revisited again.  In any case, can we pleasa type the iteration variable?

```suggestion
        await foreach (ThreadMessage message in client.GetMessagesAsync(threadId, new() { Order = MessageCollectionOrder.Descending }, cancellationToken).ConfigureAwait(false))
```